app/upgrades/v4/upgrade_test.go

```diff
@@ -185,6 +187,33 @@ func (s *UpgradeTestSuite) validateRollappsMigration(numRoll int) error {
 	return nil
 }
 
+// validate rollapp gauges
+func (s *UpgradeTestSuite) validateRollappGaugesMigration() error {
+	rollappMap := make(map[string]struct{}) // Create a map to store rollapps
+
+	rollapps := s.App.RollappKeeper.GetAllRollapps(s.Ctx)
+	for _, rollapp := range rollapps {
+		rollappMap[rollapp.RollappId] = struct{}{} // Store rollapp in the map
+	}
+
+	gauges := s.App.IncentivesKeeper.GetActiveGauges(s.Ctx)
+	if len(gauges) == len(rollapps) {
+		return fmt.Errorf("rollapp gauges not created for all rollapps")
```

Comment:
should this not be "rollapp gauges already created for all rollapps" ?
also, I would still iterate and lookup all the gauges' rollapps IDs in the `rollappMap` map.